quinn-proto/src/endpoint.rs

```diff
@@ -597,15 +597,51 @@ impl Endpoint {
         trace!(self.ctx.log, "connection got packet"; "connection" => %self.connections[conn.0].local_id, "len" => packet.payload.len());
         let was_closed = self.connections[conn.0].state.as_ref().unwrap().is_closed();
 
+        let mut err_to_log: Option<ConnectionError> = None;
         // State transitions
         let state = self.connections[conn.0].state.take().unwrap();
-        let state = self.connections[conn.0].handle_connected_inner(
+        let state = match self.connections[conn.0].handle_connected_inner(
             &mut self.ctx,
             now,
             remote,
             packet,
             state,
-        );
+        ) {
+            Ok(state) => match state {
+                State::Closed(closed) => {
+                    let copy = closed.clone();
+                    err_to_log = Some(closed.reason.into());
```

Comment:
We don't want to emit a new `ConnectionLost` event every time we receive a packet.